packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,199 @@
+            attempts++;
+            throw new Error('Failed');
+        };
+
+        try {
+            await expect(funcToRetry).to.retry({ retries: 3 });
+
+            throw new Error('This should not be called');
+        } catch (error) {
+            expect(error).to.be.an.instanceOf(Error);
+            expect((error as Error).message).to.includes('Limit of 3 retries');
+            expect(attempts).to.equal(3);
+        }
+    });
+
+    describe('options should work correctly:', () => {
+        it('timeout after the specified duration', async () => {
+            const funcToRetry = async () => {
+                await new Promise((resolve) => setTimeout(resolve, 500));
+                return 'Success';
+            };
+
+            await expect(funcToRetry)
+                .retry({ retries: 2, timeout: 250 })
+                .then(
+                    () => {
+                        throw new Error('This should not be called');
+                    },
+                    (error: Error) => {
+                        expect(error).to.be.an.instanceOf(Error);
+                        expect(error.message).to.equal('timed out after 250ms');
+                    }
+                );
+        });
+
+        it('throw an error when retries limit exceeded', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
+                attempts++;
+
+                return attempts;
+            };
+
+            await expect(funcToRetry)
+                .retry({ retries: 10 })
+                .to.be.above(100)
+                .then(
+                    () => {
+                        throw new Error('This should not be called');
+                    },
+                    (error: Error) => {
+                        expect(error).to.be.an.instanceOf(Error);
+                        expect(error.message).to.equal('Limit of 10 retries exceeded!');
+                    }
+                );
+        });
+
+        it('should retry a function that succeeds and assert the result with getter property', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
+                attempts++;
+                if (attempts < 3) {
+                    throw new Error('Failed');
+                }
+                return { success: true };
+            };
+
+            await expect(funcToRetry).retry({ retries: 5, delay: 10 }).and.have.property('success').and.be.true;
+        });
+    });
+
+    describe('should work with negated assertions:', () => {
+        it('assert object with a property `status` that does not match a specific value', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
+                attempts++;
+                return { status: attempts === 3 ? 'success' : 'pending' };
+            };
+
+            await expect(funcToRetry)
+                .retry({ retries: 5, delay: 10 })
+                .and.have.property('status')
```

Comment:
add another test and switch negative and positive assersions

Reply:
@olehrakwix added such test and it wrks according to native chai's logic that I mentioned above 
<img width="915" alt="image" src="https://user-images.githubusercontent.com/124504800/229995745-d38c5199-b3a6-4333-ac98-c8f7b550f24a.png">